zebra-chain/src/transaction/serialize.rs

```diff
@@ -267,28 +332,11 @@ impl ZcashDeserialize for Transaction {
                 let lock_time = LockTime::zcash_deserialize(&mut reader)?;
                 let expiry_height = block::Height(reader.read_u32::<LittleEndian>()?);
                 let value_balance = (&mut reader).zcash_deserialize_into()?;
-                let mut shielded_spends = Vec::zcash_deserialize(&mut reader)?;
-                let mut shielded_outputs = Vec::zcash_deserialize(&mut reader)?;
+                let shielded_spends = Vec::zcash_deserialize(&mut reader)?;
+                let shielded_outputs = Vec::zcash_deserialize(&mut reader)?;
                 let joinsplit_data = OptV4Jsd::zcash_deserialize(&mut reader)?;
-
-                use futures::future::Either::*;
-                let shielded_data = if !shielded_spends.is_empty() {
-                    Some(ShieldedData {
-                        first: Left(shielded_spends.remove(0)),
-                        rest_spends: shielded_spends,
-                        rest_outputs: shielded_outputs,
-                        binding_sig: reader.read_64_bytes()?.into(),
-                    })
-                } else if !shielded_outputs.is_empty() {
-                    Some(ShieldedData {
-                        first: Right(shielded_outputs.remove(0)),
-                        rest_spends: shielded_spends,
-                        rest_outputs: shielded_outputs,
-                        binding_sig: reader.read_64_bytes()?.into(),
-                    })
-                } else {
-                    None
-                };
+                let shielded_data =
```

Comment:
`Transaction::v4` isn't changing so I don't think we need to change this.

Reply:
The idea is to use the same code in V4 and V5 as they are doing almost the same for sapling stuff(except for the order of the fields).

Reply:
Even though they have the same names, the underlying types of these fields are different, so it might be best to just duplicate some code.